src/framework/components/element/text-element.js

```diff
@@ -672,9 +673,18 @@ Object.assign(pc, function () {
             asset.off("remove", this._onFontRemove, this);
         },
 
+        _onFontRender: function () {
+            // if the font has been changed (e.g. canvasfont re-render)
+            // re-applying the same font updates character map and ensures
+            // everything is up to date.
+            this.font = this._font;
```

Comment:
I've been trying to get away from this way of re-applying properties. Do you think maybe we could separate the part that does the applying into a separate function and call that from here instead?